6211 Батраков Владислав Александрович Лаб.4 Вар.2

Trig9ger opened the pull request:
ФИО: Батраков Владислав Александрович
Номер группы: 6211
Номер лабораторной: 4
Номер варианта: 2
Краткое описание решаемой задачи: Согласно варианту 2 лабораторной 4 мне нужно было создать датасет со столбцом "Ширина" с реализацией фильтрации и сортировки датасетов, а так же график на основе 3 полученых датасетов.
Краткое описание работы кода: Модуль "main.py" выполняет всю работу программы. Функция csv_check возвращает массив путей до изображений. Функция create_data создает датасет со столбцом Ширина.

для запуска программы нужно ввести в путь_до_аннотации три_пути_для_датасетов путь_для_графика

Trig9ger committed:
Add files via upload

Trig9ger committed:
залил 2 и 3 лабараторные работы

Trig9ger committed:
залил 2 и 3 лабараторные работы

Trig9ger committed:
испраил расположение W и H в parse_size_range

Trig9ger committed:
доработал 4 лабу

Trig9ger committed:
доработал 4 лабу

Trig9ger committed:
доработал 4 лабу

Trig9ger committed:
доработал 4 лабу

microalan commented on py/pyproj4/main.py:
```diff
@@ -0,0 +1,130 @@
+
+    parser.add_argument("annotation", type=str, help="абсолютный путь до файла анотации")
+    parser.add_argument("dataframe_paths", nargs=3, type=str, help="путь для установки измененого изображения")
+    parser.add_argument("graph_path", type=str, help="размер нового изображения формата widthXheight")
```
Поправьте help

microalan commented on py/pyproj4/main.py:
```diff
@@ -0,0 +1,130 @@
+    return filtered
+
+
+def save_datas(d: pd.DataFrame, s: pd.DataFrame, f: pd.DataFrame, paths: list) -> None:
```
Сделать для одного датафрейма

microalan commented on py/pyproj4/main.py:
```diff
@@ -0,0 +1,130 @@
+    f.to_csv(paths[2], index=True)
+
+
+def install_graphs(d: pd.DataFrame, s: pd.DataFrame, f: pd.DataFrame, path: str) -> None:
```
Сделать для одного датафрейма

microalan reviewed (commented):
1. Разделить на модули
2. Pandas умеет работать с csv, датафрейм можно создавать проще

Trig9ger committed:
разделил на модули

так же изменил метод создания датафреймов через библиотеку matplotlib а не через модуль csv. изменил функцию install_graphs на установку единичного графика. убрал функцию установки 3 датафреймов. теперь нужно вводить 3 пути для установки графиков на основе датафреймов.

microalan commented on py/pyproj4/dfhelp.py:
```diff
@@ -0,0 +1,77 @@
+    """
+    Создает DataFrame по аннотации
+    """
+    column_names = ['Абсолютеный путь', 'Относительный путь']
```
Неправильные названия, правила такие же как при именовании переменной

microalan commented on py/pyproj4/dfhelp.py:
```diff
@@ -0,0 +1,77 @@
+    False - возвращет все строки с шириной меньше value
+    """
+    if equation:
+        filtered = data[data["Ширина"] > value]
```
Неправильное название

microalan commented on py/pyproj4/main.py:
```diff
@@ -0,0 +1,36 @@
+        data['Ширина'] = dfhelp.get_widths(absolute_paths)
+
+        sorted_d = data.sort_values("Ширина", ascending=False)  # уже существует функция сортировки
+        filtered_d = dfhelp.filter_data(data, 500, True)
```
Задать ширину через аргументы командной строки

microalan commented on py/pyproj4/dfhelp.py:
```diff
@@ -0,0 +1,77 @@
+
+def parsing() -> argparse.ArgumentParser:
+    """
+    осушествляет парсинг аргументов
```
осуШествляет 😭

Trig9ger committed:
исправил наиминования столбцов, добавил параметр ширины во ввод командной строки

microalan reviewed (approved):
Зачтено
